src/logic/outfits.ts

```diff
@@ -0,0 +1,61 @@
+import { Outfits } from '@dcl/schemas'
+import { createNamesOwnershipChecker } from '../ports/ownership-checker/names-ownership-checker'
+import { createWearablesOwnershipChecker } from '../ports/ownership-checker/wearables-ownership-checker'
+import { AppComponents, TypedEntity } from '../types'
+
+export async function getOutfits(
+  components: Pick<AppComponents, 'metrics' | 'content' | 'theGraph' | 'config' | 'fetch' | 'ownershipCaches'>,
+  ethAddress: string
+): Promise<TypedEntity<Outfits> | undefined> {
+  const outfitsEntities: TypedEntity<Outfits>[] = await components.content.fetchEntitiesByPointers([ethAddress])
+
+  if (!outfitsEntities || outfitsEntities.length === 0) {
+    return undefined
+  }
+
+  const outfitsEntity = outfitsEntities[0]
+
+  if (!outfitsEntity.metadata || outfitsEntity.metadata.outfits.length === 0) {
+    return outfitsEntities[0]
+  }
+
+  const outfits = outfitsEntities[0].metadata as Outfits
+
+  const wearablesOwnershipChecker = createWearablesOwnershipChecker(components)
+  const outfitsWearables = outfits.outfits.map((outfit) => outfit.outfit.wearables).flat()
+  wearablesOwnershipChecker.addNFTsForAddress(ethAddress, outfitsWearables)
+  wearablesOwnershipChecker.checkNFTsOwnership()
+
+  const ownedWearables = new Set(wearablesOwnershipChecker.getOwnedNFTsForAddress(ethAddress))
+  const outfitsWithOwnedWearables = outfits.outfits.filter((outfit) => {
+    const outfitWearables = outfit.outfit.wearables
+    return outfitWearables.every((wearable) => ownedWearables.has(wearable))
+  })
+
+  const namesOwnershipChecker = createNamesOwnershipChecker(components)
+  namesOwnershipChecker.addNFTsForAddress(ethAddress, outfits.namesForExtraSlots)
+  namesOwnershipChecker.checkNFTsOwnership()
+  const ownedNames = new Set(namesOwnershipChecker.getOwnedNFTsForAddress(ethAddress))
+  let extraSlots = ownedNames.size
+
+  const outfitsWithOwnedNamesAndWearables = outfitsWithOwnedWearables.filter((outfit) => {
```

Comment:
I think another way to write this would be to `filter((outfit) => outfit.slot > 4)` and then compare the length of this result to  `ownedNames.size`

Reply:
The thing is not all the extra outfits are removed.
If you have 4 extra outfits, but only 2 out of 4 names are owned, the first 2 extra slots are left, and the others 2 removed.

Reply:
not quite sure to follow, can you give me an example?

Reply:
Given
```
outfits = {
  outfits: [outfitSlot1, outfitSlot3, outfitSlot5, outfistSlot8]
  names: ['name1', 'name2']
```
Then checking that `ownedNames = ['name1']`, the result should be:
```
outfits = {
  outfits: [outfitSlot1, outfitSlot3, outfitSlot5]
  names: ['name1']
}
```
The last extra slot is removed.

Anyways, just applied a refactor in the last commit, see if that helps